Feature/school 25

Flictivity opened the pull request.

Flictivity committed:
SCHOOL-25
Добавлено окно создания нового преподавателя
Добавлена фильтрация дат

Flictivity committed:
SCHOOL-25
Изменен размер окна создания преподавателя
Изменен стиль кнопки для открытия окна

WingiM commented on DumaemSchool.BlazorWeb/Pages/TeachersPage.razor:
```diff
@@ -74,14 +76,17 @@
 
+        if (res)
+        {
+            Snackbar.Add("Перподаватель успешно уволен", Severity.Success);
```
перпод

WingiM replied:
а еще у нас учителя а не преподаватели

WingiM commented on DumaemSchool.Database/Mappers/SqlUtility.cs:
```diff
@@ -41,7 +41,7 @@ public static class SqlUtility
         switch (filter.Value)
         {
-            case int integerValue:
+            case int or double or DateTime:
```
проверь на датах

WingiM commented on DumaemSchool.BlazorWeb/Shared/Windows/TeacherCreateWindow.razor:
```diff
@@ -0,0 +1,50 @@
+        var result = await Sender.Send(addTeacherCommand);
+        if (!result.IsSuccess)
+        {
+            Snackbar.Add("Не удалось добавить преподавателя!", Severity.Error);
```
там можно вытащить текст ошибки

Flictivity replied:
на англ?

Flictivity committed:
SCHOOL-25
Исрпавлен warning с non-null value

WingiM commented on DumaemSchool.BlazorWeb/Pages/TeachersPage.razor:
```diff
@@ -132,6 +137,25 @@
+
+        if (!result.Canceled)
+        {
+            Snackbar.Add("Преподаватель успешно добавлен!", Severity.Success);
```
учитель

WingiM commented on DumaemSchool.BlazorWeb/Pages/TeachersPage.razor:
```diff
@@ -132,6 +137,25 @@
+            MaxWidth = MaxWidth.Medium, FullWidth = true
+        };
+        
+        var dialog = await DialogService.ShowAsync<TeacherCreateWindow>("Новый преподаватель", options);
```
учитель

Flictivity committed:
SCHOOL-25
Изменены названия преподаватель --> учитель
Исправлена ошибка создания пользователя для учителя
